### PR TITLE
Follow POM convention to ordering POM elements + missing deployment repositories + approval tests

### PR DESCRIPTION
**Pull request by murdos**
- Follow POM convention to ordering POM elements
- Add missing snapshots and releases policies for deployment repositories
- Setup approval tests to track significant behavior changes. The following 3 cases are tested:
  - writing a complex model to a new file
  - updating a complex model to an existing file containing a very minimal model
  - updating a complex model to the file containing the complex model

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,6 +48,11 @@
             <version>2.9.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.approvaltests</groupId>
+            <artifactId>approvaltests</artifactId>
+            <version>22.3.2</version>
```
Comment (gastaldi) [inline pom.xml:54]: It's missing the `test` scope

**Review by gastaldi — CHANGES_REQUESTED** [review]
Thank you for the PR! Never heard of this `approvaltests` library before, but it belongs to the `test` scope AFAIU

File: pom.xml (hunk 1 above)
Comment (murdos) [reply pom.xml:54]: Yes, of course! it's fixed

File: src/test/java/io/fabric8/maven/MavenTest.java [hunk 2]
```diff
@@ -120,6 +121,47 @@ void should_preserve_parent_relative_path(@TempDir Path tempDir) throws Exceptio
...
+        Maven.writeModel(model, newPom);
+
+        Approvals.verify(Files.readString(newPom));
+    }
+
+    @Test
+    void should_save_full_pom_on_updated_full_file(@TempDir Path tempDir) throws Exception {
+        URL resource = getClass().getResource("full-pom.xml");
+        Path basePom = Paths.get(resource.toURI());
+        Model model = Maven.readModel(new FileReader(basePom.toFile()));
```
Comment (gastaldi) [inline src/test/java/io/fabric8/maven/MavenTest.java:140]:
```suggestion
        Model model = Maven.readModel(basePom);
```

File: src/test/java/io/fabric8/maven/MavenTest.java [hunk 3]
```diff
@@ -120,6 +121,47 @@ void should_preserve_parent_relative_path(@TempDir Path tempDir) throws Exceptio
...
+        Maven.writeModel(model, updatedPom);
+
+        Approvals.verify(Files.readString(updatedPom));
+    }
+
+    @Test
+    void should_save_full_pom_on_updated_minimal_file(@TempDir Path tempDir) throws Exception {
+        URL resource = getClass().getResource("full-pom.xml");
+        Path basePom = Paths.get(resource.toURI());
+        Model model = Maven.readModel(new FileReader(basePom.toFile()));
```
Comment (gastaldi) [inline src/test/java/io/fabric8/maven/MavenTest.java:153]:
```suggestion
        Model model = Maven.readModel(basePom);
```

**Review by gastaldi — COMMENTED** [review]
Some minor nitpicks to the test

**Comment by gastaldi** [comment]
Thank you for the PRs! I'll try them out and merge them if all looks good

**Review by gastaldi — APPROVED** [review]

**Pull request merged**